components/solr-access-service/api/src/test/java/org/phenotips/solr/HPOScriptServiceTest.java

```diff
@@ -0,0 +1,328 @@
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.params.CommonParams;
+import org.junit.Assert;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.params.SolrParams;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+
+
+import org.mockito.internal.matchers.CapturingMatcher;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.phenotips.vocabulary.SolrVocabularyResourceManager;
+import org.xwiki.cache.Cache;
+import org.xwiki.cache.CacheException;
+import org.xwiki.cache.CacheManager;
+import org.xwiki.cache.config.CacheConfiguration;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.script.service.ScriptService;
+import org.xwiki.test.mockito.MockitoComponentMockingRule;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyCollectionOf;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.anyMap;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ *  Tests for the {@link HPOScriptService} component
+ */
+public class HPOScriptServiceTest
+{
+
+    @Rule
+    public MockitoComponentMockingRule<ScriptService> mocker =
+        new MockitoComponentMockingRule<ScriptService>(HPOScriptService.class);
+
+    public HPOScriptService service;
+
+    @Mock
+    private SolrClient server;
+
+    @Mock
+    private Cache<SolrDocument> cache;
+
+    @Mock
+    private QueryResponse response;
+
+    private SolrDocumentList solrDocList;
+
+    private int indexReturn;
+
+    private final String fieldNames = "id,name,def,comment,synonym,is_a";
+
+    private final String testIDValue = "HP:0000118";
+
+    @Before
+    @SuppressWarnings("unchecked")
+    public void setUp() throws ComponentLookupException, IOException, SolrServerException, CacheException
+    {
+        MockitoAnnotations.initMocks(this);
+        this.indexReturn = -1;
+
+        SolrVocabularyResourceManager externalServicesAccess =
+                this.mocker.getInstance(SolrVocabularyResourceManager.class);
+        when(externalServicesAccess.getSolrConnection()).thenReturn(this.server);
+
+        CacheManager cacheFactory = this.mocker.getInstance(CacheManager.class);
+        when(cacheFactory.createNewLocalCache(any(CacheConfiguration.class))).thenReturn((Cache) this.cache);
```

Comment:
Remove the `@SuppressWarnings("unchecked")` everywhere in this file, and use this following style of mocking objects over whate you're doing to prevent the warning from showing up:

```java
// what you do
when(cacheFactory.createNewLocalCache(any(CacheConfiguration.class))).thenReturn((Cache) this.cache); 
// what you should do
Mockito.doReturn(this.cache).when(cacheFactory).createNewLocalCache(any(CacheConfiguration.class));
```